File: 0728-self-dividing-numbers/0728-self-dividing-numbers.py
class Solution:
    def selfDividingNumbers(self, left: int, right: int) -> List[int]:
        st=[]
        for i in range(left,right+1):
            temp, d = i, True
            while temp>0:
                r = temp%10
                if r==0 or i%r!=0:
                    d=False
                    break
                temp//=10
            if d: st.append(i)
        return st

File: 0728-self-dividing-numbers/0728-self-dividing-numbers.java
import java.util.List;
import java.util.ArrayList;
class Solution {
    public boolean isSD(int num){
        int temp = num;
        while (num > 0) {
            int rem = num % 10;
            if (rem == 0 || (temp % rem) != 0) {
                return false;
            }
            num /= 10;
        }
        return true;
    }
    public List<Integer> selfDividingNumbers(int left, int right) {
        ArrayList<Integer> ans = new ArrayList<>();
        for(int i=left; i<=right;i++){
            if(isSD(i)){
                ans.add(i);
            }
        }
        return ans;
    }
}